### DISCUSSION TITLE
Performance: One collection with 10M points vs 10 collections with 1M points

### DISCUSSION THREAD

**Discussion by alexpausan** [qdrant/qdrant#2914] [Q&A] [answered]
Hi, great work on Qdrant.
I have a question about the way to store data. 

Basic use case: for each user, I'll have approx 10M vectors. Each vector will be part of 1 category, and I have 10 categories. My question is, how to store those vectors? In 10 different collections? Or one collection would work almost the same? (And I'll rely on filters for the category, I saw the benchmark with and without filters, and I understand that penalty.)

I am trying to understand from what point on the search becomes considerably slower (with and without the filters). Would it work similarly for 50M too? From my understanding of ANN, I think it shouldn't be such a big drop, but I may be wrong. Vector size is 768. 

Thanks!

**Comment by timvisee** [comment, marked as answer]
> My question is, how to store those vectors? In 10 different collections? Or one collection would work almost the same?

We generally advice to store in one collection. That should work perfectly fine if you define the appropriate payload to separate them using filters when searching.

Please see: https://qdrant.tech/documentation/guides/multiple-partitions/

> I am trying to understand from what point on the search becomes considerably slower (with and without the filters). Would it work similarly for 50M too?

It should not get noticeably slower. Having many collections instead induces memory overhead, which might have a bigger effect on performance than putting all of it in a single collection (though 10 collections is not a lot).

Please note that we test Qdrant to work with billions of vectors.

You can always set up a test instance and define a target RPS/latency to see if the results are satisfactory.
